File: base/android/javatests/src/org/chromium/base/profiler/TestSupport.java
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.base.profiler;

import org.jni_zero.CalledByNative;
import org.jni_zero.JNINamespace;
import org.jni_zero.NativeMethods;

/** Helper to run code through JNI layer to test JNI unwinding. */
@JNINamespace("base")
public final class TestSupport {
    @CalledByNative
    public static void callWithJavaFunction(long context) {
        TestSupportJni.get().invokeCallbackFunction(context);
    }

    @NativeMethods
    interface Natives {
        void invokeCallbackFunction(long context);
    }
}

File: mojo/public/rust/tests/test_support.cc
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Similar to support.cc, this patches over changes from the old Mojo SDK to
// Mojo now. Test-specific helpers are here.

#include "mojo/public/rust/tests/test_support.h"

#include <cstdint>

#include "base/command_line.h"
#include "mojo/core/embedder/embedder.h"

void InitializeMojoEmbedder(std::uint32_t argc, const char* const* argv) {
  // Some mojo internals check command line flags, so we must initialize it
  // here.
  base::CommandLine::Init(argc, argv);
  mojo::core::Init();
}
